hudi-cli/src/main/java/org/apache/hudi/cli/commands/HoodieLogFileCommand.java

```diff
@@ -225,7 +225,7 @@ storage, new StoragePath(logFilePathPattern)).stream()
     if (shouldMerge) {
       System.out.println("===========================> MERGING RECORDS <===================");
       HoodieMergedLogRecordScanner scanner =
-          HoodieMergedLogRecordScanner.newBuilder()
+          (HoodieMergedLogRecordScanner) HoodieMergedLogRecordScanner.newBuilder()
```

Comment:
Note for reviewers: Need to cast because I moved the builder to the base class to enahnce reuse.

Reply:
Let's add the builder impl for each sub-class, just like what we do to original `HoodieUnMergedLogRecordScanner` and `HoodieMergedLogRecordScanner`.